scala/scala-impl/test/org/jetbrains/plugins/scala/lang/refactoring/anonymousToInner/ScalaAnonymousToInnerTest.scala

```diff
@@ -0,0 +1,233 @@
+        |  def hasNext = input.hasNext
+        |
+        |  def next = someNum
+        |}""".stripMargin
+
+    val className = "IntIterator"
+
+    doTest(before, after, className)
+  }
+
+  def testNestedAnonymousClasses(): Unit = {
+    val before =
+      s"""def parse(input: Iterator[Byte], sliceSize: Int): Option[Iterator[Byte]] =
+         |  if (input.hasNext) {
+         |    Some(new Iterator[Byte] {
+         |      val secondLevelIterator = new Iter${Caret}ator[Int] {
+         |
+         |        override def hasNext: Boolean = true
+         |
+         |        override def next(): Int = sliceSize
+         |      }
+         |
+         |      def hasNext = input.hasNext
+         |
+         |      def next = sliceSize.toByte
+         |    })
+         |  } else {
+         |    None
+         |  }""".stripMargin
+    val after =
+      """def parse(input: Iterator[Byte], sliceSize: Int): Option[Iterator[Byte]] =
+        |  if (input.hasNext) {
+        |    Some(new Iterator[Byte] {
+        |      val secondLevelIterator = new IntIterator(sliceSize)
+        |
+        |      def hasNext = input.hasNext
+        |
+        |      def next = sliceSize.toByte
+        |    })
+        |  } else {
+        |    None
+        |  }
+        |
+        |class IntIterator(sliceSize: Int) extends Iterator[Int] {
+        |
+        |  override def hasNext: Boolean = true
+        |
+        |  override def next(): Int = sliceSize
+        |}""".stripMargin
+
+    val className = "IntIterator"
+
+    doTest(before, after, className)
+  }
+
+  private def doTest(initialText: String, expectedText: String, className: String): Unit = {
+    scalaFixture.configureFromFileText(initialText)
+
+    val editor = CommonDataKeys.EDITOR.getData(DataManager.getInstance().getDataContextFromFocusAsync.blockingGet(5, TimeUnit.SECONDS))
+    val caretModel = editor.getCaretModel
+    assert(caretModel.getCaretCount == 1, "Expected exactly one caret.")
+    assert(caretModel.getOffset > 0, s"Not specified caret marker in test case. Use <caret> in scala file for this.")
+
+    val scalaFile = getFile.asInstanceOf[ScalaFile]
+
+    val element = scalaFile.findElementAt(caretModel.getOffset)
+    val newTemplateDefinition = Option(PsiTreeUtil.getParentOfType(element, classOf[ScNewTemplateDefinition])).getOrElse(
+      throw new Exception("expected to find the `new` keyword before the anonymous type")
+    )
+
+    val extendsBlock = newTemplateDefinition.extendsBlock
+    val (variables, targetContainer) = ScalaAnonymousToInnerHandler.parseInitialExtendsBlock(extendsBlock)
+    ScalaAnonymousToInnerHandler.performRefactoring(getProject, className, variables, extendsBlock, newTemplateDefinition, targetContainer)
+
+    Assert.assertEquals(expectedText, scalaFile.getText)
+  }
+
+  override protected def supportedIn(version: ScalaVersion): Boolean = version.isScala3
```

Comment:
Please move it to the beginning of the class, before all the methods in the test class